dummy/dummy.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// Package dummy provides dummy implementations of interfaces required, but not
+// actually used by Ethereum transaction execution. Although these are used in
+// production and aren't test doubles, they effectively fill the same role as a
+// dummy as described in https://martinfowler.com/bliki/TestDouble.html.
+package dummy
+
+import (
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/consensus"
+	"github.com/ava-labs/libevm/core"
+	"github.com/ava-labs/libevm/core/types"
+)
+
+// ChainContext returns a dummy that returns [Engine] when its Engine() method
+// is called, and panics when its GetHeader() method is called.
+func ChainContext() core.ChainContext {
+	return chainContext{}
+}
+
+// Engine returns a dummy that panics when its Author() method is called.
+func Engine() consensus.Engine {
+	return engine{}
+}
+
+type (
+	chainContext struct{}
+	engine       struct{ consensus.Engine }
+)
+
+func (chainContext) Engine() consensus.Engine                    { return engine{} }
+func (chainContext) GetHeader(common.Hash, uint64) *types.Header { panic("unimplemented") }
```

Comment:
Is this deadcode? It seems to me that this can currently be hit by the `BLOCKHASH` opcode.

Reply:
Yup, you're right. Fixed and added extensive testing of contextual op codes.